python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -39,28 +40,33 @@ class AssetSelection(ABC):
             AssetSelection.groups("marketing") & AssetSelection.keys("leads").downstream()
     """
 
+    @public  # type: ignore
     @staticmethod
     def all() -> "AllAssetSelection":
```

Comment:
these types should be added to the top level API perhaps a la `AllAssetSelection`

Reply:
my impulse would be to keep them private - they're kind of an implementation detail. though maybe by that reasoning the return types of these functions should just be AssetSelection?